### DISCUSSION TITLE
Adding Eyecandy to your Quarkus App with React and Patternfly

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#40924] [Quarkus Blog/Website]
# Adding Eyecandy to your Quarkus App with React and Patternfly

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/gui-react-patternfly/

**Comment by shakilmith** [comment]
Do I have to then host backend and react frontend in separate host? Or, only single hosting providers will serve both?

**Reply by maxandersen** [reply]
frontendmaven plugin bundles it inside the same app so it can be served out of the same process/port.

But you can obviously still choose to host the frontend and backend separately.

Note: This blog is from 2021 and since then Quinoa and Webbundler showed up which provides a deeper integration and more developer joy.  We did a Quarkus Insights on lots of these areas recently you can find at https://www.youtube.com/watch?v=C9JDsIrcTWI

**Reply by shakilmith** [reply]
Thanks for replying. I checked by myself later, I built my react app for pr and copy the the files from build directory and paste them in the Static folder of my backend app (not wise choice though), and after little modification, I saw everything is running backend + frontend in the same port. Yes, I heard about frontendmaven plugin, and prepare to use it in my production grade applications. Thanks
